docs/rules/force-types-on-object-props.md

```diff
@@ -0,0 +1,93 @@
+---
+pageClass: rule-details
+sidebarDepth: 0
+title: vue/force-types-on-object-props
+description: enforce user to add type declaration to object props
+---
+# vue/force-types-on-object-props
+
+> enforce user to add type declaration to object props
+
+- :exclamation: <badge text="This rule has not been released yet." vertical="middle" type="error"> ***This rule has not been released yet.*** </badge>
+- :gear: This rule is included in all of `"plugin:vue/base"`, `"plugin:vue/essential"`, `"plugin:vue/vue3-essential"`, `"plugin:vue/strongly-recommended"`, `"plugin:vue/vue3-strongly-recommended"`, `"plugin:vue/recommended"` and `"plugin:vue/vue3-recommended"`.
+
+## :book: Rule Details
+
+Prevent missing type declaration of not primitive objects in a TypeScript projects.
+
+Bad:
+
+<eslint-code-block :rules="{'vue/force-types-on-object-props': ['error']}">
+
+```js
+export default {
+  props: {
+    prop: {
+      type: Object,
+    },
+  },
+}
+```
+
+</eslint-code-block>
+
+<eslint-code-block :rules="{'vue/force-types-on-object-props': ['error']}">
+
+```ts
+export default {
+  props: {
+    prop: {
+      type: Array
+    }
+  }
+}
+```
+
+</eslint-code-block>
+
+Good:
+
+<eslint-code-block :rules="{'vue/force-types-on-object-props': ['error']}">
+
+```ts
+export default {
+  props: {
+    prop: {
+      type: Object as Props<Anything>,
+    }
+  }
+}
+```
+
+</eslint-code-block>
+
+<eslint-code-block :rules="{'vue/force-types-on-object-props': ['error']}">
+
+```ts
+export default {
+  props: {
+    prop: {
+      type: String, // or any other primitive type
+    }
+  }
+}
+```
+
+</eslint-code-block>
```

Comment:
This can be merged into one example:

````suggestion
<eslint-code-block :rules="{'vue/force-types-on-object-props': ['error']}">

```ts
export default {
  props: {
    prop: {
      // ✗ BAD
      type: Object,
      type: Array,
      
      // ✓ GOOD
      type: Object as Props<Anything>,
      type: String, // or any other primitive type
    }
  }
}
```

</eslint-code-block>
````